src/components/measureList/MeasureList.tsx

```diff
@@ -206,7 +236,54 @@ export default function MeasureList(props: {
     });
   };
 
-  const createVersion = () => {};
+  const createVersion = async (versionType: string) => {
+    if (
+      versionType !== "major" &&
+      versionType !== "minor" &&
+      versionType !== "patch"
+    ) {
+      setCreateVersionDialog({
+        open: true,
+        measureId: targetMeasure.current?.id,
+      });
+      setOptionsOpen(false);
+    } else {
+      await measureServiceApi
+        .createVersion(targetMeasure.current?.id, versionType)
+        .then(async () => {
+          handleDialogClose();
+          await props.onListUpdate();
+          setSnackBar({
+            message: "New version of measure is Successfully created",
+            open: true,
+            severity: "success",
+          });
+        })
+        .catch((error) => {
+          handleDialogClose();
+          const errorData = error?.response;
+          if (errorData?.status === 400) {
+            setSnackBar({
+              message: "Requested measure cannot be versioned",
+              open: true,
+              severity: "error",
+            });
+          } else if (errorData?.status === 403) {
+            setSnackBar({
+              message: "User is unauthorized to create a version",
+              open: true,
+              severity: "error",
+            });
+          } else {
+            setSnackBar({
+              message: errorData?.message,
+              open: true,
+              severity: "error",
+            });
+          }
+        });
+    }
```

Comment:
This does not cover all error instances. An example is a 500 error or the service being down for versioning. In that case there will just be an empty box that appears. You can use a fallback error message on 279 to display in case there is no errorData?.message